File: FreeTypeUnity/Assets/script/font/BTree.cs
using System;
using System.Collections.Generic;
using UnityEngine;

public class PicNode<KEY>
{
    public int width;
    public int height;
    public KEY key = default(KEY);
    // 是否旋转
    public bool IsRot = false;
}

public class BTreeNode<KEY>
{
    // 开始位置
    public int x, y;
    public PicNode<KEY> virtualPic;
    public bool isRoute;
    public bool isFull;
    public BTreeNode<KEY> left, right;
}

File: dataStruct/dataStruct/dataStruct/struct/BTree.m
//
//  BTree.m
//  dataStruct
//
//  Created by guoxiaodong on 14-8-14.
//  Copyright (c) 2014年 chinasofti. All rights reserved.
//

#import "BTree.h"

@implementation picNode
-(id)initWithSize:(CGSize)size name:(NSString*)name
{
    self = [super init];
    _size = size;
    _name = name;
    return self;
}
- (id)copyWithZone:(NSZone *)zone
{
    return self;
}
-(NSString*) description{
    return [NSString stringWithFormat:@"rect(%.0f,%.0f),name=%@",self.size.width,self.size.height,self.name];
}
@end

@implementation BTreeNode
-(NSString*) description{
    return [NSString stringWithFormat:@"point(%.0f,%.0f),isfull=%d,%@",self.point.x,self.point.y,self.isFull,self.virtualPic];
}
@end

